src/iamService.ts

```diff
@@ -0,0 +1,411 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import * as gax from './gax';
+import {GrpcClient, GrpcClientOptions, ClientStubOptions} from './grpc';
+import {GrpcClient as FallbackGrpcClient} from './fallback';
+import {createApiCall} from './createApiCall';
+import {GoogleAuth, GoogleAuthOptions, OAuth2Client} from 'google-auth-library';
+import {ProjectIdCallback} from 'google-auth-library/build/src/auth/googleauth';
+import * as routingHeader from './routingHeader';
+
+import {
+  LongrunningDescriptor,
+  PageDescriptor,
+  StreamDescriptor,
+  BundleDescriptor,
+} from './descriptor';
+import * as gapicConfig from './iam_policy_service_client_config.json';
+import * as protosTypes from '../protos/iam_service';
+import * as fallback from './fallback';
+import * as path from 'path';
+let version = require('../../package.json').version;
+
+interface Descriptors {
```

Comment:
All these interfaces are already defined in gax, can we just import them?

Reply:
yes, they are defined in `src/index.ts` but `src/index.ts` also imports `iam_service.ts` for `iamClient`, and that will result in circular dependency. (realized this circular dependency until I met some wired issues) Not sure if we have other ways to do that?

Reply:
Yes, let's move those interfaces from `src/index.ts` to `src/clientInterfaces.ts` or something like that, reexport in `src/index.ts` for client libraries, and import them from `src/iamServices.ts`.

Reply:
great idea! will do!

Reply:
done. thanks!